core-libs/setup/ssr/optimized-engine/rendering-strategy-resolver.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Request } from 'express';
+import {
+  RenderingBlocked,
+  RenderingStrategy,
+} from './ssr-optimization-options';
+
+const hasExcludedParams = (
+  request: Request,
+  renderingBlocked: RenderingBlocked
+) => {
+  const params = request.query ? Object.getOwnPropertyNames(request.query) : [];
+
+  if (!renderingBlocked?.params) {
+    return false;
+  }
+
+  return renderingBlocked.params.some((excludedParam) =>
+    params.some((param) => excludedParam === param)
```

Comment:
`could:` consider to use e.g. `indexOf` to to break the loop ASAP instead of going through the whole list each time.

Reply:
I believe `some` does break the loop.

Reply:
Yes, sorry! You're right. Sorry for the confusion 👍